### PR TITLE
rocm-related: 5.3.0 -> 5.3.1 & aggregate

### PR DESCRIPTION
**Pull request by Madouura**
###### Description of changes
Most hashes did not change, version bump to maintain compatibility.
Tracking: #197885

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Flakebi — APPROVED** [review]
Thanks again!

I tested some OpenCL and hip applications and they worked fine.

**Comment by LunNova** [comment]
Result of `nixpkgs-review pr 198770` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>hip</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>rocclr</li>
    <li>rocm-cmake</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocm-smi</li>
    <li>rocm-thunk</li>
    <li>rocminfo</li>
  </ul>
</details>

**Review by LunNova — APPROVED** [review]

**Comment by Madouura** [comment]
@lovesegfault Rolled into this PR, since they all rely on this PR anyway.

**Review by lovesegfault — APPROVED** [review]
LGTM

Will wait on @Flakebi's review to merge

**Comment by Madouura** [comment]
Rebuilt everything to make sure this PR is safe to merge.

Final result:
```console
nixpkgs on  pr/aggr [$] took 42m42s 
❯ nix-build . -A llvmPackages_rocm.llvm -A llvmPackages_rocm.clang -A hip -A rocclr -A rocm-comgr -A rocm-device-libs -A rocm-opencl-runtime -A rocm-runtime -A rocm-thunk -A rocm-cmake -A rocminfo -A rocm-smi -A rocprim -A hipcub -A rocsparse -A hipsparse -A rccl -A rocfft -A tensile -A rocblas -A miopengemm -A composable_kernel -A clang-ocl -A rocthrust
/nix/store/qn2f45hc8qi4214ss1dcmrxp5vl0iaw1-rocm-llvm-5.3.1
/nix/store/4nvx5fla45198skmxihg83wlw0k8rr57-rocm-llvm-wrapper-5.3.1
/nix/store/9mpxwijw14zkzlz2icalfkh6pb408zmq-hip-5.3.1
/nix/store/sss84hpsbfgm2xd9va9fjb23x7dfwrnj-rocclr-5.3.1
/nix/store/6vaagd85p5zwgmrfyraz1pgirbpaj73r-rocm-comgr-5.3.1
/nix/store/cf91952wf8dy4fjmjyqi5as41gjb3kc2-rocm-device-libs-5.3.1
/nix/store/a47gbycs0lfzrbfhd6s67q65xsq13rn2-rocm-opencl-runtime-5.3.1
/nix/store/9b0rir8yazpb40b1gm2xf5ljyn7ymn34-rocm-runtime-5.3.1
/nix/store/zbj0m25c60v73fr20g113m783xdgrppi-rocm-thunk-5.3.1
/nix/store/d5x36ns4lg734wkbyqbdb6ab7iyi5vda-rocm-cmake-5.3.1
/nix/store/ybbbxlxgwqp4y2jgdkjr1ddjsys9364g-rocminfo-5.3.1
/nix/store/2q6il1lcanja2w39w26r15hxxrzip287-rocm-smi-5.3.1
/nix/store/hgxqja6cjbrirzx46m8cwyx4jdy6v884-rocprim-2.11.0-5.3.1
/nix/store/s66c7hw4fgb8vc80qyjplcl6shjdxfk5-hipcub-2.12.0-5.3.1
/nix/store/a6xnva6sihlh9dav373j54zkhwzdjd3v-rocsparse-2.3.2-5.3.1
/nix/store/9l231q7bwndn2parilii8w164j6ry6c1-hipsparse-2.3.1-5.3.1
/nix/store/jp0gnv8m96m2zd8yy5ayfv4mv16gjd6x-rccl-2.12.10-5.3.1
/nix/store/80biz86ywzh6q1sxvs4xi1j72rywg185-rocfft-1.0.18-5.3.1
/nix/store/3ak9ks0bbxdp20ph0jiyvy5s2519jg0l-python3.10-tensile-4.34.0-5.3.1
/nix/store/xv3dshh1fsbr0yb41qhmrhivy7x57z17-rocblas-2.45.0-5.3.1
/nix/store/cpgzc3n4h9lml6n0y6kfc5hwqvw88l8r-miopengemm-5.3.1
/nix/store/6y2b4y2a94vqv8g1dfzkga7x9r1r1j74-composable_kernel-unstable-2022-11-02
/nix/store/mickbgzbca3x0ckcc3nirmj8vm7lq3yf-clang-ocl-5.3.1
/nix/store/lm5jm9mj548fcjknf3n5b7hyrlifffv8-rocthrust-2.16.0-5.3.1
```

File: pkgs/development/libraries/rocsparse/default.nix [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  };
+in stdenv.mkDerivation rec {
+  pname = "rocsparse";
+  rocmVersion = "5.3.1";
+  version = "2.3.2-${rocmVersion}";
+
+  outputs = [
+    "out"
+  ] ++ lib.optionals (buildTests || buildBenchmarks) [
+    "test"
```
Comment (Flakebi) [inline pkgs/development/libraries/rocsparse/default.nix:40]: How can I run the tests? I get
```bash
$ result-test/bin/rocsparse-test
"hipErrorNoBinaryForGpu: Unable to find code object for all current devices!"
fish: Job 1, 'result-test/bin/rocsparse-test' terminated by signal SIGABRT (Abort)
```
I guess it either needs something in my environment or I have to specify the GPU somewhere?

**Review by Flakebi — COMMENTED** [review]
Thanks, just two questions inline.

File: pkgs/development/libraries/rocsparse/default.nix [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+assert (buildTests || buildBenchmarks) -> gtest != null;
+assert (buildTests || buildBenchmarks) -> boost != null;
+assert (buildTests || buildBenchmarks) -> python3Packages != null;
+
+let
+  matrices = lib.optionalAttrs (buildTests || buildBenchmarks) import ./deps.nix {
+    inherit fetchzip;
+    mirror1 = "https://sparse.tamu.edu/MM";
+    mirror2 = "https://www.cise.ufl.edu/research/sparse/MM";
+  };
```
[the remark this answers is not in the document]
Comment (Madouura) [reply pkgs/development/libraries/rocsparse/default.nix:31]: That's a good idea, I'll add it to a TODO list in the tracking issue.
If for some reason hipsparse begins using different matricies, as long as some of them are the same we can still do this as well.

File: pkgs/development/libraries/rocsparse/default.nix (hunk 1 above)
Comment (Madouura) [reply pkgs/development/libraries/rocsparse/default.nix:40]: Probably yes, see: https://github.com/Madouura/nixos-config/blob/0e4236c18a468abd300b3df5c2657dba6f763ec0/hosts/machines/ura/modules/system/hardware.nix#L6-L9

**Comment by vcunat** [comment]
`rocfft` produces 3.6 GiB output.  That's too large, so Hydra.nixos.org always fails after finishing the costly build.  Unless you can improve this, I suggest adding `meta.hydraPlatforms = [];` so that this pointless build isn't repeated there every time without succeeding.

**Comment by Madouura** [comment]
AFAIK it can't be improved, it's just a ridiculously large build.
IIRC, it's all in ``/lib`` too.
I'll submit a PR for it.

**Comment by Madouura** [comment]
@vcunat #200705

**Comment by vcunat** [comment]
Yes, I looked before, it's gigabytes of so-libraries.

**Pull request merged**